### PR TITLE
Machine learning release notes 10.15

### PR DESCRIPTION
**Pull request by VRAN016**

### CODE REVIEW

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+---
+
+### Improvements in Machine Learning Engine
+
+- Enhancements:
+  - In the Machine Learning web application, confirmation popups have been modified to provide a consistent UX with the rest of the platform.
+
+* Upgrades:
+  * Machine Learning web application upgraded to use Cumulocity Web SDK 1015.0.2.
+  * Zementis microservice upgraded from openssl  version 1.0.2 to  version 1.1.1n.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:14]:
```suggestion
  * Zementis microservice upgraded from openssl version 1.0.2 to version 1.1.1n.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  * Nyoka and Onnx microservices upgraded NumPy from version 1.18.5 to version 1.21.5.
+  * Nyoka and Onnx microservices upgraded scikit-learn from version 0.21.3 to version 0.23.1.
+  * Onnx microservice upgraded from onnx version 1.7.0 to version 1.11.0
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:17]: Either use a fullstop at the end everywhere or nowhere. Please unify.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  * Onnx microservice upgraded from onnx-runtime version 1.5.2 to version 1.11.0
+  * Onnx microservice upgraded from protobuf version 3.17.0 to version 3.20.0.
+
+- Bug fixes:
+  - Fixed the issue where requests to Onnx microservice were failing if User Agent validation was enabled in the OAI settings.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:22]:
```suggestion
  - Fixed the issue where requests to the Onnx microservice were failing if user agent validation was enabled in the OAI settings.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  - Fixed the issue where model group list was displaying the deleted model groups at certain times.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:23]:
```suggestion
  - Fixed the issue where the model group list was displaying the deleted model groups at certain times.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  - Fixed the issue where listed pmml models reported 404-Not Found while attempting to delete them.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:24]:
```suggestion
  - Fixed the issue where listed pmml models reported "404-Not Found" while attempting to delete them.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+
+### Improvements in Machine Learning Workbench
+
+* Compute off-load with a pluggable Enterprise Gateway:\
+With this release, the Cumulocity IoT Machine Learning Workbench provides an integration connection to Jupyter Enterprise Gateway using which it is possible to launch kernels on behalf of remote notebooks. The enterprise gateway exposes a Kernel as a Service model providing better optimization of compute resources, improved multi-user support, enable compute offload capability and more granular security for Jupyter notebook environment.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:30]:
```suggestion
With this release, the Cumulocity IoT Machine Learning Workbench provides an integration connection to Jupyter Enterprise Gateway which allows to launch kernels on behalf of remote notebooks. The Enterprise Gateway exposes a kernel as a service model providing better optimization of compute resources, improved multi-user support, enable compute offload capability and more granular security for the Jupyter notebook environment.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  - Settings: 
+    - A new tab is added to register the Jupyter Enterprise Gateway url
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:32]:
```suggestion
    - A new tab is added to register the Jupyter Enterprise Gateway URL

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    - Pop-up to notify an end user that all running kernels would be deleted during the setting and delete actions of the JEG credentials
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:33]:
```suggestion
    - Popup to notify an end user that all running kernels will be deleted during the setting and delete actions of the JEG credentials

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  - Notebook UI: 
+      - Pop-up display to select different kernel specs
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:35]:
```suggestion
      - Popup display to select different kernel specs

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+      - “Jupyter” logo beside the Kernel name on every notebook to notify an end user that the notebook is running using a Kernel spec from the remote Jupyter Enterprise Gateway server
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:36]:
```suggestion
      - “Jupyter” logo beside the kernel name on every notebook to notify the user that the notebook is running using a kernel spec from the remote Jupyter Enterprise Gateway server

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+      - Option within the integrated Jupyter UI to insert the code snippets using which a Data Scientist would be able to:
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:37]:
```suggestion
      - Option within the integrated Jupyter UI to insert the code snippets which allow a data scientist to:

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+          - Download “Data” from MLW to the JEG server
+          - Upload “Models” to MLW from JEG server
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:39]:
```suggestion
          - Upload “Models” to MLW from the JEG server

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+          - List the projects available
+          - List the resources available within a project
+
+* MLW sdk:\
+With the inclusion of Jupyter Enterprise Gateway connection within the Cumulocity IoT Machine Learning Workbench, there was a need to develop a Python package using which the artifacts (Data, Models) exchange was possible to and from the JEG server.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:44]:
```suggestion
With the inclusion of the Jupyter Enterprise Gateway connection within the Cumulocity IoT Machine Learning Workbench, there was a need to develop a Python package which allowed to exchange the artifacts (Data, Models) to and from the JEG server.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+- Enhancements:
+  - Project Management UX improvements:
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:47]:
```suggestion
  - Project management UX improvements:

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+    - Ability to select multiple resources for deletion
+    - Ability to delete individual project version
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:49]:
```suggestion
    - Ability to delete an individual project version

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+    - Ability to select multiple projects for deletion
+    - Tasks, Assets grouping according to the projects
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:51]:
```suggestion
    - Tasks, assets grouping according to the projects

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+    - Ability to select multiple tasks, assets for deletion
+    - Project commit button is restructured to make the functionality more intuitive
+
+* Upgrades:
+  * Machine Learning Workbench web application upgraded to use Cumulocity Web SDK 1015.0.2.
+  * MLW microservice upgraded from onnx version 1.7.0 to version 1.11.0.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:57]:
```suggestion
  * MLW microservice upgraded from Onnx version 1.7.0 to version 1.11.0.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+  * MLW microservice upgraded from protobuf version 3.12.4 to version 3.20.0.
+  * MLW microservice upgraded from supervisor version 4.2.1 to version 4.2.4.
+
+- Bug fixes:
+  - Fixed the issue with vulnerable 3rd Party Component protobuf used in docker image.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:62]:
```suggestion
  - Fixed the issue with vulnerable 3rd-party component protobuf used in docker image.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+  - Fixed the issue with vulnerable 3rd Party Component openssl used in docker image.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:63]:
```suggestion
  - Fixed the issue with the vulnerable 3rd-party component openssl used in the Docker image.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+  - Fixed the issue with the wrong resource count for a particular project.
+  - Fixed the issue with MLW RBAC not working if user agent validation required is checked in Cumulocity Authentication (OAuth) Settings.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:65]:
```suggestion
  - Fixed the issue with MLW RBAC not working if "user agent validation required" is checked in the Cumulocity IoT authentication settings (OAuth).

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 1 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:14]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:17]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 3 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:22]: fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 4 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:23]: fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 5 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:24]: fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 20 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:65]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 19 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:63]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 18 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:62]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 17 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:57]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 16 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:51]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 15 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:49]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 14 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:47]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 13 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:44]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 12 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:39]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 10 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:36]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 9 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:35]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 8 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:33]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 7 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:32]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+
+### Improvements in Machine Learning Engine
+
+- Enhancements:
+  - In the Machine Learning web application, confirmation popups have been modified to provide a consistent UX with the rest of the platform.
+
+* Upgrades:
+  * Machine Learning web application upgraded to use Cumulocity Web SDK 1015.0.2.
+  * Zementis microservice upgraded from openssl version 1.0.2 to version 1.1.1n.
+  * Nyoka and Onnx microservices upgraded NumPy from version 1.18.5 to version 1.21.5.
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:15]:
```suggestion
  * Nyoka and Onnx microservices upgraded from NumPy version 1.18.5 to version 1.21.5.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+  * Nyoka and Onnx microservices upgraded scikit-learn from version 0.21.3 to version 0.23.1.
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:16]:
```suggestion
  * Nyoka and Onnx microservices upgraded from scikit-learn version 0.21.3 to version 0.23.1.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+  * Onnx microservice upgraded from onnx version 1.7.0 to version 1.11.0.
+  * Onnx microservice upgraded from onnx-runtime version 1.5.2 to version 1.11.0.
+  * Onnx microservice upgraded from protobuf version 3.17.0 to version 3.20.0.
+
+- Bug fixes:
+  - Fixed the issue where requests to the Onnx microservice were failing if user agent validation was enabled in the OAI settings.
+  - Fixed the issue where the model group list was displaying the deleted model groups at certain times.
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:23]:
```suggestion
  - Fixed the issue where the model group list was displaying the deleted model groups occasionally.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+  - Fixed the issue where listed pmml models reported "404-Not Found" while attempting to delete them.
+
+
+### Improvements in Machine Learning Workbench
+
+* Compute off-load with a pluggable Enterprise Gateway:\
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:29]:
```suggestion
* Compute off-load with a pluggable Jupyter Enterprise Gateway (JEG):\

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+With this release, the Cumulocity IoT Machine Learning Workbench provides an integration connection to Jupyter Enterprise Gateway which allows to launch kernels on behalf of remote notebooks. The Enterprise Gateway exposes a kernel as a Service model providing better optimization of compute resources, improved multi-user support, enable compute offload capability and more granular security for Jupyter notebook environment.
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:30]:
```suggestion
With this release, the Cumulocity IoT Machine Learning Workbench provides an option to connect to a Jupyter Enterprise Gateway which enables launch and management of kernels on behalf of remote notebooks. The JEG exposes kernel-as-a-service model providing better optimization of compute resources, improved multi-user support, enable compute offload capability and more granular security for Jupyter notebook environment.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+  - Settings: 
+    - A new tab is added to register the Jupyter Enterprise Gateway URL
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:32]:
```suggestion
    - A new tab is added to register the JEG URL

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+    - Popup to notify the user that all running kernels will be deleted during the setting and delete actions of the JEG credentials
+  - Notebook UI: 
+      - Popup display to select different kernel specs
+      - “Jupyter” logo beside the kernel name on every notebook to notify the user that the notebook is running using a kernel spec from the remote Jupyter Enterprise Gateway server
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:36]:
```suggestion
      - “Jupyter” logo beside the kernel name on every notebook to notify the user that the notebook is running using a kernel spec from the remote JEG server

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+      - Option within the integrated Jupyter UI to insert the code snippets which allow a data scientist to:
+          - Download “Data” from MLW to the JEG server
+          - Upload “Models” to MLW from the JEG server
+          - List the projects available
+          - List the resources available within a project
+
+* MLW sdk:\
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:43]:
```suggestion
* MLW SDK:\

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+With the inclusion of the Jupyter Enterprise Gateway connection within the Cumulocity IoT Machine Learning Workbench, there was a need to develop a Python package which allowed to exchange the artifacts (Data, Models) to and from the JEG server.
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:44]:
```suggestion
To complement the inclusion of the Jupyter Enterprise Gateway connection within the Cumulocity IoT Machine Learning Workbench, we also provide a Python package which facilitates exchange of artifacts (Data, Models, Resources) to and from the JEG server.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 30]
```diff
@@ -0,0 +1,65 @@
...
+
+- Enhancements:
+  - Project management UX improvements:
+    - Ability to select multiple resources for deletion
+    - Ability to delete an individual project version
+    - Ability to select multiple projects for deletion
+    - Tasks, assets grouping according to the projects
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:51]:
```suggestion
    - Task and Asset grouping based on projects

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 31]
```diff
@@ -0,0 +1,65 @@
...
+    - Ability to select multiple tasks, assets for deletion
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:52]:
```suggestion
    - Ability to select multiple tasks and assets for deletion

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 32]
```diff
@@ -0,0 +1,65 @@
...
+    - Project commit button is restructured to make the functionality more intuitive
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:53]:
```suggestion
    - Project commit button relocated to make the functionality more intuitive

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 33]
```diff
@@ -0,0 +1,65 @@
...
+
+* Upgrades:
+  * Machine Learning Workbench web application upgraded to use Cumulocity Web SDK 1015.0.2.
+  * MLW microservice upgraded from Onnx version 1.7.0 to version 1.11.0.
+  * MLW microservice upgraded from protobuf version 3.12.4 to version 3.20.0.
+  * MLW microservice upgraded from supervisor version 4.2.1 to version 4.2.4.
+
+- Bug fixes:
+  - Fixed the issue with the vulnerable 3rd Party component protobuf used in the Docker image.
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:62]:
```suggestion
  - Fixed the issue with the vulnerable 3rd Party component protobuf.

```
Can we also link the CVE ticket?

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 34]
```diff
@@ -0,0 +1,65 @@
...
+  - Fixed the issue with the vulnerable 3rd Party component openssl used in the Docker image.
```
Comment (maulingajjar) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:63]:
```suggestion
  - Fixed the issue with the vulnerable 3rd Party component openssl.

```
Can we also link the CVE ticket?

Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:63]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 32 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:53]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 31 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:52]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 30 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:51]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 29 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:44]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 28 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:43]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 27 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:36]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 26 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:32]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 25 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:30]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 24 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:29]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 23 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:23]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 22 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:16]: Fixed.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md (hunk 21 above)
Comment (VRAN016) [reply content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:15]: Fixed.

**Review by maulingajjar — APPROVED** [review]
Requested changes incorporated.

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 35]
```diff
@@ -0,0 +1,66 @@
...
+weight: 40
+---
+
+### Improvements in Machine Learning Engine
+
+- Enhancements:
+  - In the Machine Learning web application, confirmation popups have been modified to provide a consistent UX with the rest of the platform.
+
+* Upgrades:
+  * Machine Learning web application upgraded to use Cumulocity Web SDK 1015.0.2.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:13]:
```suggestion
  * Machine Learning web application upgraded to use Cumulocity IoT Web SDK 1015.0.2.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 36]
```diff
@@ -0,0 +1,66 @@
...
+- Bug fixes:
+  - Fixed the issue where requests to the Onnx microservice were failing if user agent validation was enabled in the OAI settings.
+  - Fixed the issue where the model group list was displaying the deleted model groups occasionally.
+  - Fixed the issue where listed pmml models reported "404-Not Found" while attempting to delete them.
+
+
+### Improvements in Machine Learning Workbench
+
+* Compute off-load with a pluggable Jupyter Enterprise Gateway (JEG):\
+With this release, the Cumulocity IoT Machine Learning Workbench provides an option to connect to a Jupyter Enterprise Gateway which enables launch and management of kernels on behalf of remote notebooks. The JEG exposes kernel-as-a-service model providing better optimization of compute resources, improved multi-user support, enable compute offload capability and more granular security for Jupyter notebook environment.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:30]:
```suggestion
With this release, the Cumulocity IoT Machine Learning Workbench provides an option to connect to a Jupyter Enterprise Gateway which enables launch and management of kernels on behalf of remote notebooks. The JEG exposes a kernel-as-a-service model providing better optimization of compute resources, improved multi-user support, enable compute offload capability and more granular security for Jupyter notebook environment.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 37]
```diff
@@ -0,0 +1,66 @@
...
+* MLW SDK:\
+To complement the inclusion of the Jupyter Enterprise Gateway connection within the Cumulocity IoT Machine Learning Workbench, we also provide a Python package which facilitates exchange of artifacts (Data, Models, Resources) to and from the JEG server.
+
+
+- Enhancements:
+  - Project management UX improvements:
+    - Ability to select multiple resources for deletion
+    - Ability to delete an individual project version
+    - Ability to select multiple projects for deletion
+    - Task and Asset grouping based on projects
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:52]:
```suggestion
    - Task and asset grouping based on projects

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 38]
```diff
@@ -0,0 +1,66 @@
...
+    - Ability to select multiple tasks and assets for deletion
+    - Project commit button relocated to make the functionality more intuitive
+
+* Upgrades:
+  * Machine Learning Workbench web application upgraded to use Cumulocity Web SDK 1015.0.2.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:57]:
```suggestion
  * Machine Learning Workbench web application upgraded to use Cumulocity IoT Web SDK 1015.0.2.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 39]
```diff
@@ -0,0 +1,66 @@
...
+
+- Enhancements:
+  - In the Machine Learning web application, confirmation popups have been modified to provide a consistent UX with the rest of the platform.
+
+* Upgrades:
+  * Machine Learning web application upgraded to use Cumulocity Web SDK 1015.0.2.
+  * Zementis microservice upgraded from openssl version 1.0.2 to version 1.1.1n.
+  * Nyoka and Onnx microservices upgraded from NumPy version 1.18.5 to version 1.21.5.
+  * Nyoka and Onnx microservices upgraded from scikit-learn version 0.21.3 to version 0.23.1.
+  * Onnx microservice upgraded from onnx version 1.7.0 to version 1.11.0.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:17]:
```suggestion
  * Onnx microservice upgraded from Onnx version 1.7.0 to version 1.11.0.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 40]
```diff
@@ -0,0 +1,66 @@
...
+  * Onnx microservice upgraded from onnx-runtime version 1.5.2 to version 1.11.0.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:18]:
```suggestion
  * Onnx microservice upgraded from ONNX-runtime version 1.5.2 to version 1.11.0.

```

File: content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md [hunk 41]
```diff
@@ -0,0 +1,66 @@
...
+    - Ability to select multiple resources for deletion
+    - Ability to delete an individual project version
+    - Ability to select multiple projects for deletion
+    - Task and Asset grouping based on projects
+    - Ability to select multiple tasks and assets for deletion
+    - Project commit button relocated to make the functionality more intuitive
+
+* Upgrades:
+  * Machine Learning Workbench web application upgraded to use Cumulocity Web SDK 1015.0.2.
+  * MLW microservice upgraded from Onnx version 1.7.0 to version 1.11.0.
```
Comment (BeateRixen) [inline content/release-10-15-0/machine-learning-10-15-0-bundle/10_15_0.md:58]:
```suggestion
  * MLW microservice upgraded from ONNX version 1.7.0 to version 1.11.0.

```

**Review by BeateRixen — APPROVED** [review]

**Comment by BeateRixen** [comment]
@VRAN016 Can we merge this?

**Pull request merged**